connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -353,9 +353,10 @@ message Sample {
   // (Optional) The random seed.
   optional int64 seed = 5;
 
-  // (Optional) Explicitly sort the underlying plan to make the ordering deterministic.
-  // This flag is only used to randomly splits DataFrame with the provided weights.
-  optional bool force_stable_sort = 6;
+  // (Required) Explicitly sort the underlying plan to make the ordering deterministic or cache it.
+  // This flag is true when invoking `dataframe.randomSplit` to randomly splits DataFrame with the
+  // provided weights. Otherwise, it is false.
+  bool deterministic_order = 6;
```

Comment:
I think it should be optional.

Reply:
The default value `false` is fine here. @amaliujia what's the principle to mark field as required or optional?

Reply:
I think the most useful principle is if you are not satisfied the default value of a field and you must know whether it is set or not: then use optional. Otherwise leave it as default field access rule.